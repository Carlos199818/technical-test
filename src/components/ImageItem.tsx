import { Box, Button, Grid, Typography } from "@mui/material";
import type { RoverPhoto } from "../interfaces/RoverPhotoInterface";


export default function ImageItem({ item, onDelete }: { item: RoverPhoto; onDelete: (imageToDelete: RoverPhoto) => void }) {


    return (
        <Grid size={4} key={item.id}>
            <Box
                sx={{
                    width: "100%",
                    height: "480px",
                    backgroundColor: "#2b2b2b",
                }}>
                <Box sx={{
                    height: "10%",
                }}>
                    <Typography variant="h6" sx={{
                        color: "#fff",
                        textAlign: "center",
                        fontWeight: "bold",
                        paddingTop: "10px",
                    }}>
                        {item.camera.full_name}
                    </Typography>
                </Box>
                <img
                    src={item.img_src}
                    alt="Mars"
                    style={{
                        width: "100%",
                        height: "80%",
                        objectFit: "cover",
                        borderRadius: "10px",
                    }}
                />
                <Box sx={{
                    height: "10%",
                }}>
                    <Button onClick={() => onDelete(item)} sx={{
                        backgroundColor: "#2b2b2b",
                        color: "#f00",
                        "&:hover": {
                            backgroundColor: "#000000",
                        },
                    }}>
                        Eliminar
                    </Button>
                </Box>
            </Box>

        </Grid>
    )
}