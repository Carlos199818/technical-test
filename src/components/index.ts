

export { default as Navbar } from "./Navbar"
export { default as BoxImages } from "./BoxImages"
export { default as ImageItem } from "./ImageItem"