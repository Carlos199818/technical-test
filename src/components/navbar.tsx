import { AppBar, Grid, Typography } from "@mui/material"


export default function Navbar({ navHeight }: { navHeight: number }) {

    return (
        <AppBar
            sx={{
                backgroundColor: "#2b2b2b",
                height: `${navHeight}px`,
            }}
        >
            <Grid container sx={{ p: 5 }}>

                <Grid size={2}>
                    <Typography>
                        Bienvenidos a Marte
                    </Typography>
                </Grid>
                <Grid size={8} sx={{ textAlign: "center" }}>
                    <Typography variant="h4">
                        Ven y disfruta de la experiencia de conocer el planeta rojo desde la comodidad de tu hogar
                    </Typography>
                </Grid>
                <Grid size={2} sx={{ textAlign: "right" }}>
                    <Typography>
                        Welcome to Mars
                    </Typography>
                </Grid>

            </Grid>

        </AppBar>
    )
} 