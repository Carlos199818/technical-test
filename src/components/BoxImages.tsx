import { Box, Button, Grid, Typography } from "@mui/material";
import { useEffect, useState } from "react";
import type { RoverPhoto } from "../interfaces/RoverPhotoInterface";
import axiosInstance from "../api/client";
import ImageItem from "./ImageItem";

export default function BoxImages() {
    const itemsPerPage: number = 12;

    const [data, setData] = useState<RoverPhoto[]>([]);
    const [visibleCount, setVisibleCount] = useState<number>(itemsPerPage);

    const handleShowMore = (): void => {
        setVisibleCount((prev: number) => prev + itemsPerPage);
    };

    const handleDelete = (imageToDelete: RoverPhoto): void => {
        setData((prevItems) => prevItems.filter((image) => image !== imageToDelete));
    };

    useEffect(() => {
        axiosInstance("/photos?sol=1000&api_key=DEMO_KEY")
            .then((response) => {
                setData(response.data.photos);
            })
            .catch((error) => {
                console.error("Error al obtener los datos:", error);
            });
    }, []);


    return (
        <Grid container spacing={5} sx={{ p: 5 }}>

            {
                data.slice(0, visibleCount).map((item: RoverPhoto) => (
                    <ImageItem
                        key={item.id}
                        item={item}
                        onDelete={handleDelete} 
                    />
                ))
            }


            <Grid size={12}>
                <Box
                    sx={{
                        width: "100%",
                        height: "70px",
                        display: "flex",
                        justifyContent: "center",
                    }}>
                    <Button variant="contained" onClick={handleShowMore} sx={{
                        backgroundColor: "#2b2b2b",
                        color: "#fff",
                        "&:hover": {
                            backgroundColor: "#000000",
                        },
                    }}>
                        <Typography variant="h6">
                            Más imágenes
                        </Typography>
                    </Button>
                </Box>
            </Grid>
        </Grid>
    )
}