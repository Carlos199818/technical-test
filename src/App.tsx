// import reactLogo from './assets/react.svg'

import { Box } from "@mui/material"
import { BoxImages, Navbar } from "./components"

function App() {

  const navHeight = 240;

  return (
    <Box sx={{
      display: "flex",
      height: "100vh",

    }}>

      <Box>
        <Navbar navHeight={navHeight} />
      </Box>

      <Box
        sx={{
          width: "100%",
          mt: `${navHeight}px`,
        }}
      >
        <BoxImages />
      </Box>
    </Box>
  )
}

export default App
