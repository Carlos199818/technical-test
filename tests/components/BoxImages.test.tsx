import { render } from '@testing-library/react';
import { BoxImages } from '../../src/components';

describe('Pruebas en <BoxImages />', () => {

    test('Debe hacer match con el snapshot', () => {

        const { container } = render(<BoxImages />);
        expect(container).toMatchSnapshot();

    });

});