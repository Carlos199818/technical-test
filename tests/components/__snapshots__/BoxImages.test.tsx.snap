// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <BoxImages /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-5 css-1g5gtdm-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-6v0m12"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1629p8p-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
          >
            Más imágenes
          </h6>
        </button>
      </div>
    </div>
  </div>
</div>
`;
